baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LogSafePreconditionsMessageFormat.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.Category;
+import com.google.errorprone.BugPattern.LinkType;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.regex.Pattern;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "LogSafePreconditionsMessageFormat",
+        category = Category.ONE_OFF,
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = LinkType.CUSTOM,
+        severity = SeverityLevel.ERROR,
+        summary = "logsafe Preconditions.checkX() methods must have '{}' style formatting.")
```

Comment:
@gracew why do we accept parameters at all?

Reply:
Seems like params are just more helpful information that devs can use to try to figure out what went wrong when a stack trace appears in their high-side logs...

why would we not want them?

Reply:
we don't inline parameters for slf4j log calls, so we would we support them here?

Reply:
Oh right, you mean just getting rid of the `{}`... yeah that seems fine to me!  As long as people can keep tacking on values to the varargs call and they end up on disk somehow :)

Reply:
The logsafe preconditions doesn't inline them either, but I do think that even if they aren't actually substituted they can be useful (particularly when named) as they are a visual indicator "you should look in the params block if this is important to you", without them you don't necessarily know if a noun is just a noun, or a placeholder for a param.

Reply:
OK, I changed the check to merely say what it was actually doing anyway, which is flagging you should not be using print-f style formatting, regardless of using any braces.

Is that fair?